enhancements/node-latency-profile/node-latency-profiles.md

```diff
@@ -0,0 +1,276 @@
+
+* User wants to fine tune the cluster reliability for their scenario while making sure they are protected from setting parameters that could potentially break the cluster.
+
+## Proposal
+
+* A new Custom Resource `NodeLatencyProfile` should be created.
+* This CR should be monitored by [MCO](https://github.com/openshift/machine-config-operator), [KCMO](https://github.com/openshift/cluster-kube-controller-manager-operator) and [CKAO](https://github.com/openshift/cluster-kube-apiserver-operator).
+* `KubeletConfigController` of MCO will modify Kubelet specific arguments
+* KCMO will modify `Controller Manager` specific arguments using `ConfigObserver`
+* CKAO will modify `Kube API Server` specific arguments using `ConfigObserver`
+
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+* Succeffully update the relavent arguments of the `Kubelet`, `Controller Manager`, `Kube API Server`
+* End user documentation
+
+#### Tech Preview -> GA
+* More testing (upgrade, downgrade, scale)
+* Optinally make it available during installation
+
+#### Removing a deprecated feature
+
+N/A
+
+## Design Details
+
+### Reliability Profile API
+
+We propose a new following [config v1 API](https://github.com/openshift/api/tree/master/config/v1) which will be consumed by the `NodeLatencyProfile`,
+
+
+```go
+type NodeLatencyProfile struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec NodeLatencyProfileSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status NodeLatencyProfileStatus `json:"status"`
+}
+```
+
+Please refer to the `Kubespray` [documentation](https://github.com/kubernetes-sigs/kubespray/blob/master/docs/kubernetes-reliability.md#recommendations-for-different-cases) for more details on different scenarios.
+```go
+type ProfileType int
+
+const (
+    // Fast Update and Fast Reaction.
+    // See - https://github.com/kubernetes-sigs/kubespray/blob/af04906b518efe8d6dcfc9e2ff57b17b49c20295/docs/kubernetes-reliability.md#fast-update-and-fast-reaction
+    FUFR ProfileType = "FUFR" //
+
+    // Medium Update and Average Reaction
+    // See - https://github.com/kubernetes-sigs/kubespray/blob/af04906b518efe8d6dcfc9e2ff57b17b49c20295/docs/kubernetes-reliability.md#medium-update-and-average-reaction
+    MUAR ProfileType = "MUAR"
+
+    // Low Update and Slow Reaction
+    // See - https://github.com/kubernetes-sigs/kubespray/blob/af04906b518efe8d6dcfc9e2ff57b17b49c20295/docs/kubernetes-reliability.md#low-update-and-slow-reaction
+    LUSR ProfileType = "LUSR"
+)
+```
+
+```go
+type NodeLatencyProfileSpec struct {
+  // NodeLatencyProfileType determins the type of NodeLatencyProfile to set on the cluster.
+  // Only follow values "FUFR", "MUAR" or "LUSR" are allowed.
+	NodeLatencyProfileType ProfileType `json:"profileType"`
```

Comment:
set a default

Reply:
Considering how much of a broad impact these settings have, I am not in favour of setting any defaults. Let user explicitly tell us which `NodeLatencyProfile` they want to set. I don't think it's safe for us to assume any default here because we could just end up breaking their cluster. 

The default value to me only makes sense when there could be a reasonable value of a parameter that could be assumed in absence of user provided one. But in this case, user `must` tell us which latency they will like the cluster to operate in. Assuming anything incorrect, like via default, could even bring down the cluster. 

IMO, user should explicity specific either `FastUpdateAndFastReaction` or `MediumUpdateAndAverageReaction` or `LowUpdateAndSlowReaction`. If nothing is provided, we just error out without taking any action.

Reply:
Although, we could introduce another profile type which has current default values for args of Kubelet, CM and KAS. Something like `DefaultUpdateAndDefaultReaction`. @sttts does that make sense?

Reply:
just call it `Default`